phpBB/phpbb/storage/controller/attachment.php

```diff
@@ -0,0 +1,497 @@
+<?php
+/**
+ *
+ * This file is part of the phpBB Forum Software package.
+ *
+ * @copyright (c) phpBB Limited <https://www.phpbb.com>
+ * @license GNU General Public License, version 2 (GPL-2.0)
+ *
+ * For full copyright and license information, please see
+ * the docs/CREDITS.txt file.
+ *
+ */
+
+namespace phpbb\storage\controller;
+
+use phpbb\auth\auth;
+use phpbb\cache\service;
+use phpbb\config\config;
+use phpbb\content_visibility;
+use phpbb\db\driver\driver_interface;
+use phpbb\event\dispatcher;
+use phpbb\exception\http_exception;
+use phpbb\request\request;
+use phpbb\storage\storage;
+use phpbb\user;
+use Symfony\Component\HttpFoundation\Request as symfony_request;
+use Symfony\Component\HttpFoundation\RedirectResponse;
+use Symfony\Component\HttpFoundation\ResponseHeaderBag;
+
+/**
+ * Controller for /download/attachment/{id} routes
+ */
+class attachment extends controller
+{
+	/** @var auth */
+	protected $auth;
+
+	/** @var config */
+	protected $config;
+
+	/** @var content_visibility */
+	protected $content_visibility;
+
+	/** @var dispatcher */
+	protected $dispatcher;
+
+	/** @var request */
+	protected $request;
+
+	/** @var user */
+	protected $user;
+
+	/**
+	 * Constructor
+	 *
+	 * @param auth					$auth
+	 * @param service				$cache
+	 * @param config				$config
+	 * @param content_visibility	$content_visibility
+	 * @param driver_interface		$db
+	 * @param dispatcher_interface	$dispatcher
+	 * @param request				$request
+	 * @param storage				$storage
+	 * @param symfony_request		$symfony_request
+	 * @param user					$user
+	 */
+	public function __construct(auth $auth, service $cache, config $config, content_visibility $content_visibility, driver_interface $db, dispatcher $dispatcher, request $request, storage $storage, symfony_request $symfony_request, user $user)
+	{
+		parent::__construct($cache, $db, $storage, $symfony_request);
+
+		$this->auth = $auth;
+		$this->config = $config;
+		$this->content_visibility = $content_visibility;
+		$this->dispatcher = $dispatcher;
+		$this->request = $request;
+		$this->user = $user;
+	}
+
+	/**
+	 * {@inheritdoc}
+	 */
+	public function handle($id)
+	{
+		$attach_id = (int) $id;
+		$thumbnail = $this->request->variable('t', false);
+
+		// Start session management, do not update session page.
+		$this->user->session_begin(false);
+		$this->auth->acl($this->user->data);
```

Comment:
this has already been done in app.php

Reply:
Only that line?

Reply:
the three of them, but session_begin is done with `ession_begin(true)` though I'm not sure it changes anything if we call it again...,,

Reply:
Yes, but the parameter of session_begin is to don't update session page, so don't know

Reply:
That's a good point. We need to be able to prevent updating the session page even when we're in a controller.